src/App.tsx

```diff
@@ -1,27 +1,65 @@
-import React from 'react';
+import {
+  useState,
+  FC,
+  useCallback,
+} from 'react';
 import './App.scss';
+
+import _ from 'lodash';
+
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
+
+type GetAllFunction = () => Promise<Good[]>;
+
+export const App: FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<Good[]>([]);
+
+  const loadGoods = useCallback(async (func: GetAllFunction) => {
+    try {
+      const goodsFromServer = await func();
+
+      setVisibleGoods((prevVisibleGoods) => {
```

Comment:
but i still make a fetch when i click in the same button
![image](https://github.com/mate-academy/react_dynamic-list-of-goods/assets/91411692/a0376902-5f31-4d0c-9ae2-798b49e49fe9)